src/FakeItEasy.Analyzer/UnusedReturnValueAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using static DiagnosticDefinitions;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class UnusedReturnValueAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly ImmutableDictionary<string, DiagnosticDescriptor> Diagnostics =
+            GetDiagnosticsMap(
+                nameof(UnusedCallSpecification));

```

Comment:
I was surprised by this, as I was thinking of the different flavours of unused return values as a single diagnostic, just with different descriptions. Shows my inexperience with the technology.

Given the multiple diagnostics, I like how you've ensured we'll need minimal changes to support a new one. :+1: